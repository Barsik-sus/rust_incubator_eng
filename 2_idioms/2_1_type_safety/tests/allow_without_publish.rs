use step_2_1::Post;

fn main()
{
  let post = Post::new( "Test".into(), "Text".into() );
  post.allow();
}