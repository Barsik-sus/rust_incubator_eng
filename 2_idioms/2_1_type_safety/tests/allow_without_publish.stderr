error[E0599]: no method named `allow` found for struct `Post<step_2_1::New>` in the current scope
 --> tests/allow_without_publish.rs:6:8
  |
6 |   post.allow();
  |        ^^^^^ method not found in `Post<step_2_1::New>`
  |
  = note: the method was found for
          - `Post<step_2_1::Unmoderated>`
